gobblin-core/src/main/java/gobblin/writer/ThrottleWriter.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.writer;
+
+import gobblin.util.limiter.Limiter;
+import gobblin.util.limiter.RateBasedLimiter;
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.State;
+import gobblin.instrumented.Instrumented;
+import gobblin.metrics.GobblinMetrics;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.codahale.metrics.Timer;
+import com.github.rholder.retry.RetryerBuilder;
+import com.google.common.base.Optional;
+
+/**
+ * Throttle writer follows decorator pattern that throttles inner writer by either QPS or by bytes.
+ * @param <D>
+ */
+public class ThrottleWriter<D> implements DataWriter<D>, Retriable {
+  private static final Logger LOG = LoggerFactory.getLogger(ThrottleWriter.class);
+  public static final String WRITER_THROTTLE_TYPE_KEY = "writer.throttle_type";
+  public static final String WRITER_LIMIT_RATE_LIMIT_KEY = "writer.throttle_rate";
+  public static final String WRITES_THROTTLED_TIMER = "gobblin.writer.throttled.time";
+
+  private static final String LOCAL_JOB_LAUNCHER_TYPE = "LOCAL";
```

Comment:
Please remove this. You can use " JobLauncherFactory.JobLauncherType.LOCAL.toString()" instead.